holodeck/agents.py

```diff
@@ -91,12 +90,20 @@ def teleport(self, location=None, rotation=None):
         val = 0
         if location is not None:
             val += 1
-            np.copyto(self._teleport_buffer, location)
+            np.copyto(self._teleport_buffer[0:3], location)
         if rotation is not None:
-            np.copyto(self._rotation_buffer, rotation)
+            np.copyto(self._rotation_buffer[3:6], rotation)
             val += 2
         self._teleport_bool_buffer[0] = val
 
+    def set_state(self, location, rotation, velocity, angular_velocity):
+        val = 4
```

Comment:
I don't think this is the value you want. Unless the implementation has changed, this should be a bit string with ones indicating that the element should be updated.
e.g.
0001 = update location (1)
0010 = update rotation (2)
0100 = update velocity (4)
1000 = update angular velocity (8)
1111 = update everything (15)

Also, for this reason, you could rewrite set_state to have default values of None and change the value based on that. Maybe we should also change the name of _teleport_bool_buffer since it doesn't quite capture what it's doing.